src/include/homestore/btree/detail/btree_remove_impl.ipp

```diff
@@ -465,26 +466,79 @@ btree_status_t Btree< K, V >::merge_nodes(const BtreeNodePtr& parent_node, const
         // Lets see if we have enough room in parent node to accommodate changes. This is needed only if the key is not
         // fixed length. For fixed length node merge will always result in lesser or equal size
         int64_t post_merge_size{0};
-        auto& old_node = old_nodes[leftmost_src.ith_nodes.back()];
-        if (old_node->total_entries()) {
-            post_merge_size += old_node->get_nth_obj_size(
-                std::min(leftmost_src.last_node_upto, old_node->total_entries() - 1)); // New leftmost entry
+        auto excess = old_nodes.size() - new_nodes.size();
```

Comment:
It is important to note that the available size for the parent node may appear to be adequate at first, but can potentially lead to space shortages during the merging process itself. To prevent such shortages, it is necessary to estimate the available space in advance, just as it would occur during the actual merge. This would enable the calculation and avoidance of any potential violations that may arise during the merging process, such as those caused by insufficient space.
the step is a following:

> 1- release space  in the parent due to excess nodes
> 2- remove old last key and insert new last key (and check the available space for each iteration)
> 3- update the left most key

Reply:
To be more precise, it is the updated keys in the parent * max_key_size(). The number of updated keys are `new_nodes.size() + 1 `.  There is a possibility that the last key of all new nodes and also the last key of the leftmost node can be updated.

Reply:
<img width="813" alt="Screenshot 2023-10-10 at 10 57 12 AM" src="https://github.com/eBay/HomeStore/assets/116847813/6735f6c4-1069-49b9-96c2-e52eac3acb81">